### PR TITLE
Classic Envs Update

### PR DESCRIPTION
**Pull request by reginald-mclean**
This update removes the use of self.dones in the Classic envs, replaces self.dones with self.terminations and self.truncations, and changes references from __was_done_step to _was_dead_step. There is also an update to pettingzoo/utils/wrappers/terminate_illegal.py that removes self.dones and uses self.terminations and self.truncations

### CODE REVIEW

File: pettingzoo/classic/chess/chess.py [hunk 1]
```diff
@@ -53,7 +53,6 @@ def __init__(self):
         }
 
         self.rewards = None
-        self.dones = None
```
Comment (jjshoots) [inline pettingzoo/classic/chess/chess.py:56]: Is there a reason this was removed but not replaced with terminations and truncations?

File: pettingzoo/classic/chess/chess.py [hunk 2]
```diff
@@ -93,21 +92,23 @@ def reset(self, seed=None, return_info=False, options=None):
...
+        self.truncations = {name: False for name in self.agents}
         self.infos = {name: {} for name in self.agents}
 
         self.board_history = np.zeros((8, 8, 104), dtype=bool)
 
     def set_game_result(self, result_val):
         for i, name in enumerate(self.agents):
-            self.dones[name] = True
+            self.terminations[name] = True
+            self.truncations[name] = True
```
Comment (jjshoots) [inline pettingzoo/classic/chess/chess.py:104]: This should be True for terminations.

**Review by jjshoots — COMMENTED** [review]
Hi @reginald-mclean, overall LGTM, aside from the couple of comments I've left. Could you also run `pip3 install black && black .` on the root directory once you're done with the fixes? This fixes the code formatting.

**Comment by jjshoots** [comment]
Hey @WillDudley, I've given this a first pass review, but do you mind giving it a second pass?

**Comment by WillDudley** [comment]
Looking at this now :)

**Review by WillDudley — APPROVED** [review]
lgtm :)

Note that the RLCard envs still need to be refactored by someone.

**Comment by jjshoots** [comment]
@WillDudley Thanks for the review, I'll have a look at RLCard in the next couple of days.
Thanks for the contribution @reginald-mclean!  (On a side note if you'd like to work on RLCard, do lmk :))

**Comment by WillDudley** [comment]
@jjshoots just looking at RLCard now, give me 30 mins

**Pull request merged**